### PR TITLE
Miscellaneous enhancements based on review comments

### PR DESCRIPTION
**Pull request by weiiwang01**
Following an internal review of the charm, several suggestions for improvement were identified. This pull request addresses those suggestions by implementing a minor refactor of the charm. No changes were made to the charm's features.

### CODE REVIEW

**Comment by gregory-schiano** [comment]
Can you write a description ?

File: src/databases.py [hunk 1]
```diff
@@ -7,10 +7,9 @@
 import pathlib
 import typing
 
-import ops.framework
 import yaml
 from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires, DatabaseRequiresEvent
-from ops.charm import CharmBase
+from ops import CharmBase, Object
```
Comment (gregory-schiano) [inline src/databases.py]: I think the recommendation we to simply import ops and use the first-level object available

**Comment by weiiwang01** [comment]
> Can you write a description ?

👍

File: tests/unit/test_charm_state.py [hunk 2]
```diff
@@ -77,13 +76,13 @@ def test_charm_state_invalid_flask_config(harness: Harness, charm_config: dict)
...
-    "charm_config, wsgi_path",
-    [
-        pytest.param({}, "app:app", id="default"),
-        pytest.param({"webserver_wsgi_path": "foo.bar:app2"}, "foo.bar:app2", id="non-default"),
-    ],
-)
+CHARM_STATE_WSGI_PATH_TEST_PARAMS = [
+    pytest.param({}, "app:app", id="default"),
+    pytest.param({"webserver_wsgi_path": "foo.bar:app2"}, "foo.bar:app2", id="non-default"),
+]
```
Comment (gregory-schiano) [inline tests/unit/test_charm_state.py]: I don't think that was required here as there aren't so many params. The comment was more for long parametrize calls

**Review by gtrkiller — DISMISSED** [review]
LGTM

File: src/databases.py [hunk 3]
```diff
@@ -49,7 +48,7 @@ def __init__(self, charm: CharmBase):
             for require in metadata["requires"].values()
             if require["interface"] in FLASK_SUPPORTED_DB_INTERFACES
         )
-
+        # automatically create database relation requirers to manage database relations
```
Comment (jdkandersson) [inline src/databases.py:51]: Perhaps expand this a bit more to indicate why a dictionary is used?

File: src/databases.py (hunk 1 above)
Comment (jdkandersson) [reply src/databases.py]: I think either is fine, it depends on how much we actually need from it and whether the namespace is useful

File: tests/unit/test_charm_state.py (hunk 2 above)
Comment (weiiwang01) [reply tests/unit/test_charm_state.py]: Updated, thanks!

File: src/databases.py (hunk 1 above)
Comment (weiiwang01) [reply src/databases.py]: I changed to use the `ops` module directly.

**Review by jdkandersson — DISMISSED** [review]

File: src/databases.py (hunk 3 above)
Comment (weiiwang01) [reply src/databases.py:51]: Added, thanks!

**Review by gregory-schiano — DISMISSED** [review]
LGTM

File: src/charm.py [hunk 4]
```diff
@@ -83,7 +73,7 @@ def __init__(self, *args: typing.Any) -> None:
         self.framework.observe(self.on.rotate_secret_key_action, self._on_rotate_secret_key_action)
         self.framework.observe(self.on.secret_storage_relation_changed, self._on_config_changed)
```
Comment (gregory-schiano) [inline src/charm.py]: This needs to be changed, it should map to a dedicated method

**Review by jdkandersson — APPROVED** [review]

File: src/charm.py (hunk 4 above)
Comment (weiiwang01) [reply src/charm.py]: Updated, thanks!

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Pull request merged**